### PR TITLE
disable extension on hires fix

### LINKED ISSUE #8: Any way to use this with Highres Fix
Hi there! Is there any way to use this with highres fix where it won't apply to the second round? When applied to the second round it makes the image blurry and slightly doubled and thus doesn't work well. Thanks again for your wonderful extension!

### PR DESCRIPTION
**Pull request by ljleb**
Disable denoising flips on hires fix. Works with batch processing.
Please make sure to test before merging @dfaker, it is possible that I have missed 1 or 2 use cases.

Closes #8

The diff might look big but the code was just dedented for better readability. This basically only adds a boolean `self.next_denoise_is_hires`.

### CODE REVIEW

**Comment by dfaker** [comment]
This'd certainly catch the simple case of a single batch image, but in turn breaks on larger batch sizes, practically I think a property on the state global set by the internals of Auto1111 is the only real solution.

**Comment by ljleb** [comment]
I tested with a batch of 4 images without any issue, can you share an example where this breaks? Just to make sure I understand the issue you are describing.

Edit: I understand the issue you are describing. Batch count > 1 is not taken into account indeed.

**Comment by ljleb** [comment]
Found a fix for batch using `p.enable_hr`. Let me know if it works on your side. I tested with batch size=2, batch count=5, xyz=3 (s/r 3 values) and it should work with batch now.

Unless extensions can call the denoising function in an unusual pattern the webui does not follow, this should work.

**Comment by dfaker** [comment]
Nice, works for me, put a toggle in to toggle the skip_hires behaviour and I'll merge this.

File: scripts/__pycache__/latent_mirroring.cpython-310.pyc
Comment (dfaker) [inline scripts/__pycache__/latent_mirroring.cpython-310.pyc:1]: Please remove this pyc file

Comment (ljleb) [reply scripts/__pycache__/latent_mirroring.cpython-310.pyc:1]: ah I don't know why I didn't see it. For sure

**Comment by ljleb** [comment]
Added checkbox in txt2img. In img2img it is not present and defaults to false in the script. (i.e. original behavior by default)

**Comment by Ame-Gura** [comment]
Disable-during-hires-path seems not working when **Maximum steps fraction to mirror at**< 1.0

```
if params.sampling_step >= params.total_sampling_steps * self.mirroring_max_step_fraction:
            self.is_hires = True
            return
```

is working in my case.

**Comment by theblacktiger59** [comment]
> Disable-during-hires-path seems not working when **Maximum steps fraction to mirror at**< 1.0
> 
> ```
> if params.sampling_step >= params.total_sampling_steps * self.mirroring_max_step_fraction:
>             self.is_hires = True
>             return
> ```
> 
> is working in my case.

Works for me too, otherwise i get blurry hires, (SDXL)

**Pull request merged**